### DISCUSSION TITLE
git@github.com: Permission denied (publickey).

### DISCUSSION THREAD

**Discussion by Srhansen849** [community/community#127820] [Programming Help] [answered]
### Body

I created a new repository and I cloned it using the HTTP, then I put my project into the folder, then did "git add ." It said files were added, then I committed and went to push using "git push origin master" It came back with "fatal: 'origin' does not appear to be a git repository fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists."
So I tried without origin " $ git push master
fatal: 'master' does not appear to be a git repository
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists."

Then I remembered that you need to set up an upstream, so I tried 
"$ git push --set-upstream origin master
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists."

So I tried generating a new ssh using the steps from https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent
but its still giving me this permission error. What is happening and why?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by interceptic** [comment]
You may need to type ``git remote -v`` to make sure you're in the right repository.

You can also type ``gh auth status`` to check if you're authenticated, to verify your ssh keys type ``ssh -T git@github.com``.

If everything here looks correct then I'm not sure, sorry :(

**Reply by Srhansen849** [reply]
Okay tried your suggestion and I'm in the right repo but your ```gh auth status``` code came back with this:
```
Angular_Projects/PortfolioProject2024 (master)
$ git remote -v
origin  git@github.com:Srhansen849/PortfolioProject.git (fetch)
origin  git@github.com:Srhansen849/PortfolioProject.git (push)

Angular_Projects/PortfolioProject2024 (master)
$ gh auth status
bash: gh: command not found
```
I'm using git bash so maybe there's a different command? and finally I tried the last command and got this.
``` 
$ ssh -T git@github.com
git@github.com: Permission denied (publickey).
```

Thank you for helping!

**Comment by eclipsology** [comment]
You can check your remote URL via `git remote -v`. If there is no remote repository, use `git remote add origin https://github.com/your-username/your-repository.git` to add your repository.

If that's not the issue, try to test whether your ssh keys are correct with `ssh -T git@github.com`

**Reply by Srhansen849** [reply]
I tried you suggestion as well and got back this
```
Angular_Projects/PortfolioProject2024 (master)
$ git remote add origin https://github.com/Srhansen849/PortfolioProject.git
error: remote origin already exists.

Angular_Projects/PortfolioProject2024 (master)
$ ssh -T git@github.com
git@github.com: Permission denied (publickey).
```
Thank you for helping!

**Reply by anko** [reply, marked as answer]
- The `git remote` command's output looks good. :+1: 

- As for `ssh`: Have you [added your SSH public key to your Github account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)? You mentioned generating the SSH keys, but not adding it to your account.

**Reply by ebndev** [reply]
Hi @anko, thanks for taking the time to answer!

@Srhansen849 if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. It would be helpful for other users who might experience this in the future 😄

**Reply by asishbhandari** [reply]
Angular_Projects/PortfolioProject2024 (master)
$ git remote add origin https://github.com/Srhansen849/PortfolioProject.git
error: remote origin already exists.

in the above line it seems like you have already set the origin, if you want to set the new origin try the following command
git remote set-url origin https://github.com/Srhansen849/PortfolioProject.git

else if you want to use the ssh, 
 step1:- generate the key
    ssh-keygen -t ed25519 -C "your_email@example.com"
    
 step 2. Copy the key:
   look for something like id_ed25519.pub. where you have generated the key, it is usually in the ./ssh folder
   Copy that entire key and go to:
  👉 [GitHub > Settings > SSH and GPG keys](https://github.com/settings/keys)
  ➕ Click "New SSH key" and paste it in.

step 3: Test it:
  ssh -T git@github.com

**Reply by HassaunB** [reply]
setting the new origin seems to have work for me thanks.

**Comment by sushant160404** [comment]
nice

**Comment by lpf-akashi** [comment]
I meet the same problem, do you fix it?

**Comment by dewcansam** [comment]
I had forked a repo then created a testing branch. I followed the steps recommended exactly but anytime I did a push to update my branch I would get this:
`error: src refspec git@github.com does not match any` trying to use HTTPS did not work either but at one point I saw some post referring to **gh auth**. I did these steps and fixed the issue.
```
sudo apt install gh
gh auth login
gh auth setup-git
gh auth status
git push https://github.com/me/myrepo.git
```
I do have a key installed. I'm not exactly sure what I was doing wrong but this is what fixed it. In case anyone else comes across this post. Maybe it can help someone else.
My setup:
OS: LMDE 6 (faye) x86_64
DE: Cinnamon 6.4.8
Shell: bash 5.2.15
Term: terminator 2.1.2
